### PR TITLE
refactor(workspace): replace static-assertions with const fn expressions

### LINKED ISSUE #3595: Replace static-assertions with const fn expressions
### File Location(s)

[static-assertions](https://github.com/nvzqz/static-assertions) crate has not been supported for the last 4 years, but as of Rust 1.57 we can do the same in a constant context: https://internals.rust-lang.org/t/nicer-static-assertions/15986

### Proposal

```diff
- static_assertions::foo!()
+ const _: () = assert!(FOO == BAR);
```

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Resolves #3595

### CODE REVIEW

File: core/src/pages.rs [hunk 1]
```diff
@@ -35,8 +35,8 @@ pub const WASM_PAGE_SIZE: usize = 0x10000;
 /// native page size, so can vary.
 pub const GEAR_PAGE_SIZE: usize = 0x4000;
 
-static_assertions::const_assert!(WASM_PAGE_SIZE < u32::MAX as usize);
-static_assertions::const_assert_eq!(WASM_PAGE_SIZE % GEAR_PAGE_SIZE, 0);
+const _: () = assert!(WASM_PAGE_SIZE < u32::MAX as usize);
+const _: () = assert!(WASM_PAGE_SIZE % GEAR_PAGE_SIZE == 0);
```
Comment (shamilsan) [inline core/src/pages.rs:39]: Is it possible to use `assert_eq` here? As it shows left and right if not equal.

**Review by shamilsan — APPROVED** [review]

File: core/src/pages.rs (hunk 1 above)
Comment (StackOverflowExcept1on) [reply core/src/pages.rs:39]: not yet: rust-lang/rust#74925

**Comment by shamilsan** [comment]
@StackOverflowExcept1on, need to appease clippy.

**Comment by StackOverflowExcept1on** [comment]
@shamilsan rust-lang/rust-clippy#11966

**Review by breathx — APPROVED** [review]
atm whitelist clippy error pls and mark them to be deleted once compiler upgraded

**Comment by StackOverflowExcept1on** [comment]
@breathx I have temporarily disabled false positive clippy lint in CI

**Review by techraed — APPROVED** [review]

**Pull request merged**